Libraries/Arduino_BNO055/BNO055_IMU.cpp

```diff
@@ -0,0 +1,138 @@
+#include <BNO055_IMU.h>
+
+void read_data(AxisData *data, DataMode mode)
+{
+    switch (mode) {
+        case GYRO:
+            {
+                uint8_t raw_data[6];
```

Comment:
Probably best to just create this raw data array outside the switch statement and use it in all cases.